ArduSub/Sub.cpp

```diff
@@ -43,7 +43,13 @@ Sub::Sub()
 #if CONFIG_HAL_BOARD != HAL_BOARD_SITL
     failsafe.pilot_input = true;
 #endif
+    if (_singleton != nullptr) {
+        AP_HAL::panic("Can only be one Sub");
+    }
```

Comment:
This should be protected by a definition to be sure this panic only occurs SITL.